src/CryptoKey.ts

```diff
@@ -0,0 +1,209 @@
+import * as core from 'webcrypto-core'
+
+import { Key as AskarKey, Jwk } from '@hyperledger/aries-askar-shared'
+import type { EcdsaParams, JsonWebKey, KeyAlgorithm, KeyFormat, KeyImportParams, KeyType, KeyUsage } from './types'
+import { askarAlgorithmToCryptoAlgorithm, cryptoAlgorithmToAskarAlgorithm } from './askar'
+import { AsnConvert, AsnParser } from '@peculiar/asn1-schema'
+import { ecdsaWithSHA256 } from '@peculiar/asn1-ecc'
+import { SubjectPublicKeyInfo } from '@peculiar/asn1-x509'
+
+export class AskarCryptoKey extends core.CryptoKey {
+  public askarKey: AskarKey
+
+  public constructor({
+    askarKey,
+    algorithm,
+    extractable = false,
+    usages,
+    type,
+  }: {
+    askarKey: AskarKey
+    extractable?: boolean
+    usages: Array<KeyUsage>
+    type?: KeyType
+    algorithm: KeyAlgorithm
+  }) {
+    super()
+    this.askarKey = askarKey
+    this.extractable = extractable
+    this.type = type
+    this.usages = usages
+    this.algorithm = algorithm
+  }
+
+  public [Symbol.dispose]() {
+    this.askarKey.handle.free()
+  }
```

Comment:
How are we intending to use this? Using `use`? Or will we manually call dispose?

Reply:
Not required anymore if we only support direct credo for now.